src/libstore/remote-store.hh

```diff
@@ -180,6 +180,8 @@ protected:
 
     virtual void narFromPath(const StorePath & path, Sink & sink) override;
 
+    Path addPermRoot(const StorePath & path, const Path & gcRoot);
+
```

Comment:
I don't think we want this here. Can we just have it in `MountedSSHStore`?

Reply:
I tried to put this into `MountedSSHStore` but wasn't able to because [the `ConnectionHandle` class](https://github.com/NixOS/nix/blob/fbc296e00fa05703ea08f9ec478a3cfe11abf3e2/src/libstore/remote-store.cc#L285) is not visible to sub-classes.

If I did this, then I would have to lift the `ConnectionHandle` class (and all the other related hidden definitions, such as [`Pool<RemoteStore::Connection>`](https://github.com/NixOS/nix/blob/fbc296e00fa05703ea08f9ec478a3cfe11abf3e2/src/libstore/remote-store.cc#L287) out into the `remote-store.hh` header.

This will be quite a change. Let me know if that's okay.

Reply:
I gave it a try and it doesn't actually look that bad. I managed to move `addPermRoot` to `ManagedSSHStore` 👍 

Please take another look.